sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserInterface.scala

```diff
@@ -33,4 +34,7 @@ trait ParserInterface {
 
   /** Creates TableIdentifier for a given SQL string. */
   def parseTableIdentifier(sqlText: String): TableIdentifier
+
+  /** Creates StructType for a given SQL string. */
+  def parseTableSchema(sqlText: String): StructType
```

Comment:
So the difference between the `parseDataType` and the `parseTableSchema` functions is that the latter allows you to parse a comma separated list of field definitions which will preserve the correct Hive metadata? Could you document this?